docs/contents/internals.rst

```diff
@@ -302,13 +302,56 @@ bcbio.yaml config::
 - step 8. Concatenate all cb-histogram-filtered.txt files::
 
     cat project-[all-barcodes]/cb-histogram-filtered.txt > cb-histogram.txt
-    
+
 Tests
 ~~~~~
 To run bcbio automated tests, install bcbio and clone bcbio master repository. You are testing your installation with tests provided in bcbio-nextgen/tests::
 
     which bcbio_nextgen.py
     cd bcbio-nextgen/tests
     ./run_tests.sh > tests.out
-    
-Tests are in integration/*.py. Each test has a set or marks. Marks are listed in pytest.ini. The mark defines how many tests to select. By default (just running plain ./run_tests.sh), it is speed1 = 11 tests. 
+
+Tests are in integration/*.py. Each test has a set or marks. Marks are listed in pytest.ini. The mark defines how many tests to select. By default (just running plain ./run_tests.sh), it is speed1 = 11 tests.
+
+Profiling
+~~~~~~~~~
+Profiling (tracking CPU, memory, IO usage) could help to optimize resource
+usage of bcbio, especially when running  on a server or AWS instance.
+Sometimes running a bcbio project with 32 cores is just 10% more efficient
+than with 16 cores, because a particular configuration might have memory or
+IO related bottlenecks.
+
+- step 1. install and start sysstat deamon: http://www.leonardoborda.com/blog/how-to-configure-sysstatsar-on-ubuntudebian/.
+- step 2. Create a cron job to gather system statistics every minute or two.
+- step 3. Before bcbio start, drop system memory caches. Otherwise memory usage statistic might be misleading::
+
+    sudo su
```

Comment:
`sudo -i` does the same thing. Also, I would add a comment like "# or become root" (not everyone uses Ubuntu or Ubuntu derivatives).